cmd/sysl/cmd_databasescript_mod.go

```diff
@@ -0,0 +1,211 @@
+	"strings"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+
+	sysl "github.com/anz-bank/sysl/pkg/sysl"
+	"github.com/sirupsen/logrus"
+)
+
+func GenerateModDatabaseScripts(datagenParams *CmdContextParamDatagen,
+	modelOld *sysl.Module, modelNew *sysl.Module, logger *logrus.Logger) (map[string]string, error) {
+	outmap := make(map[string]string)
+
+	logger.Debugf("project: %v\n", datagenParams.project)
+	logger.Debugf("title: %s\n", datagenParams.title)
+	logger.Debugf("filter: %s\n", datagenParams.filter)
+	logger.Debugf("output: %s\n", datagenParams.output)
+
+	spclass := constructFormatParser("", datagenParams.classFormat)
+
+	// The "project" app that specifies the data models to be built
+	var appOld, appNew *sysl.Application
+	var existsOld, existsNew bool
+	appOld, existsOld = modelOld.GetApps()[datagenParams.project]
+	_, existsNew = modelOld.GetApps()[datagenParams.project]
+	if !(existsOld || existsNew) {
+		return nil, fmt.Errorf("project not found in sysl for either new or old script")
+	}
+
+	epname := "Relational-Model"
+	endptOld := appOld.GetEndpoints()[epname]
+	endptNew := appNew.GetEndpoints()[epname]
+	outputDir := datagenParams.output
+	if strings.Contains(outputDir, "%(epname)") {
+		of := MakeFormatParser(datagenParams.output)
+		outputDir = of.FmtOutput(datagenParams.project, epname, "", nil)
+	}
+	generateModDatabaseScripts(spclass, outmap, modelOld, modelNew, endptOld.GetStmt(), endptNew.GetStmt(),
+		datagenParams.title, datagenParams.project, outputDir)
+	return outmap, nil
+}
+
+func generateModDatabaseScripts(pclass ClassLabeler, outmap map[string]string, modOld *sysl.Module, modNew *sysl.Module,
+	stmtsOld []*sysl.Statement, stmtsNew []*sysl.Statement, title, project, outDir string) {
+	appsOld := modOld.GetApps()
+	appsNew := modNew.GetApps()
+	var stringBuilder strings.Builder
+	v := MakeEmptyDatabaseScriptView()
+
+	// Parse all the applications in the project
+	for _, stmt := range stmtsOld {
+		if a, ok := stmt.Stmt.(*sysl.Statement_Action); ok {
+			appOld := appsOld[a.Action.Action]
+			appNew := appsNew[a.Action.Action]
+			// add/retained in the new sysl file
+			if appOld != nil && appNew != nil {
+				typeMapOld := appOld.GetTypes()
+				typeMapNew := appNew.GetTypes()
+				tableDepthMapOld := v.createTableDepthMap(typeMapOld)
+				tableDepthMapNew := v.createTableDepthMap(typeMapNew)
+
+				tablesWithActions := findAddedDeletedRetainedTables(typeMapOld, typeMapNew, tableDepthMapOld, tableDepthMapNew)
+				processTables(tablesWithActions, title, project, outDir, pclass, stringBuilder, outmap)
+			}
+			// app deleted from the new sysl file
+			if appOld != nil && appNew == nil {
+				//typeMapOld := appOld.GetTypes()
+			}
+		}
+	}
+	for _, stmt := range stmtsNew {
+		if a, ok := stmt.Stmt.(*sysl.Statement_Action); ok {
+			appNew := appsNew[a.Action.Action]
+			appOld := appsOld[a.Action.Action]
+			// app added in the new sysl file
+			if appNew != nil && appOld == nil {
+				//typeMapNew := appNew.GetTypes()
```

Comment:
The code needs cleanup I see a lot of commented. Also please ensure that the linter is run before the code is checked in.